build-system/tasks/changelog.js

```diff
@@ -200,81 +268,150 @@ function buildSections(githubMetadata) {
 
 /**
  * Get the latest git tag from either a normal release or from a canary release.
- * @return {!Promise<string>}
+ * @param {!GitMetadataDef} gitMetadata
+ * @return {!Promise<GitMetadataDef>}
  */
-function getLastGitTag() {
+function getLastGitTag(gitMetadata) {
   var options = {
     args: `describe --abbrev=0 --first-parent --tags ${branch}`
   };
   return gitExec(options).then(function(tag) {
-    return tag.trim();
+    if (!tag) {
+      throw new Error('Could not find latest ' + branch + ' tag.');
+    }
+    gitMetadata.tag = tag.trim();
+    util.log(util.colors.green('Current latest tag: ' + gitMetadata.tag));
+    return gitMetadata;
   });
 }
 
 /**
+ * Runs `git log ${branch}...{tag} --pretty=oneline --first-parent`
  * @param {string} tag
- * @return {!Promise<string>}
+ * @return {!Promise<GitMetadataDef>}
  */
-function getGitLog(tag) {
+function getGitLog(gitMetadata) {
   var options = {
-    args: 'log ' + branch + '...' + tag +
-        ' --pretty=format:%s --merges --first-parent'
+    args: `log ${branch}...${gitMetadata.tag} --pretty=oneline --first-parent`
   };
-  return gitExec(options).then(function(log) {
-    if (!log) {
-      throw new Error('No log found "git log ' + branch +
+  return gitExec(options).then(function(logs) {
+    if (!logs) {
+      throw new Error('No logs found "git log ' + branch +
           '...' + tag + '".\nIs it possible that there is no delta?\n' +
           'Make sure to fetch and rebase (or reset --hard) the latest ' +
           'from remote upstream.');
     }
-    return log;
+    const commits = logs.split('\n').filter(log => !!log.length);
+    gitMetadata.logs = commits.map(log => {
+      const words = log.split(' ');
+      return { sha: words.shift(), title: words.join(' ') };
+    });
+    return gitMetadata;
   });
 }
 
-function fetchGithubMetadata(ids) {
-  var options = {
-    url: 'https://api.github.com/repos/ampproject/amphtml/pulls/',
-    headers: {
-      'User-Agent': 'amp-changelog-gulp-task'
-    }
-  };
+/**
+ * @param {!GitMetadataDef} gitMetadata
+ * @return {!Promise<!GitMetadataDef>}
+ */
+function getGithubPullRequestsMetadata(gitMetadata) {
+  // (erwinm): Github seems to only return data for the first 3 pages
+  // from my manual testing.
+  return BBPromise.all([
+      getClosedPullRequests(1),
```

Comment:
haha yeah, it actually already did for one commit (wasn't in the first 3 pages). we do still fall back below to fetching /pulls/${id} if we can parse the id from the commit message and if nothing else just use the commit message w/o PR link. for some reason github returns no data with page 4 and up. (used curl)